framework/src/outputs/PNGTest.C

```diff
@@ -0,0 +1,279 @@
+//
+//   std::string PNGFile = "Adam";
+//
+//   PNGFile += appe;
+//   PNGFile += ".png";
+
+  numpng++;
+
+//  BoundingBox box = MeshTools::create_bounding_box(*_mesh_ptr);
+
+  //FindBBox fbbx;
+  //std::cout << "The min and the max " << (comm().min(box.min()))(0) << comm().max(box.max()) << "\n";
+
+//  _mesh_ptr->comm(); ----------- communicator we have access to.
+//  comm(); ------------------ Another communicator we have access to.
+  //_app._comm ----- this is the communicator.  It's protected.
+
+
+
+  BoundingBox boxy = _mesh_ptr->getInflatedProcessorBoundingBox(.1);
+
+  Point ptx = box.max();
+  Point ptn = box.min();
+
+  //_mesh_ptr->comm().min(ptn);
+
+  Real distx = ptx(0) - ptn(0);
+  Real disty = ptx(1) - ptn(1);
+
+  std::cout << "Values: " << distx << " " << disty << "\n";
+
+  std::string PNGFile2 = PNGFile;
+  PNGFile2 += std::to_string((int)numpng);
+
+  FILE * fp = nullptr;
+  png_structrp pngp = nullptr;
+  png_infop infop = nullptr;
+  png_bytep row = nullptr;
+//    int status = 1;
+  Real depth = 8;
+//  Real resolution = 3000;
+  // Real xdist = 2;
+  // Real ydist = 2;
+  Real width = distx*resolution;
+  Real height = disty*resolution;
+
+  if(picked != -1) {
+    if(numpng != picked) {
+      std::cout << "Not the picked one.";
+      return;
+    }
+    std::cout << "The picked one." << PNGFile2;
+  }
+
+  else
+    std::cout << "Nothing selected.";
+
+//  std::cout << "\n\n\nFile opening!\n\n\n";
+  fp = fopen(PNGFile2.c_str(), "wb");
+//  std::cout << "\n\n\nFile OPENED!\n\n\n";
+
+  if(!fp) {
+    std::cout << "Failed to open.";
+    return;
+  }
+
+  pngp = png_create_write_struct(PNG_LIBPNG_VER_STRING, nullptr, nullptr, nullptr);
+  if(!pngp) {
+    std::cout << "Failed to make pointer string.";
+    return;
+  }
+
+  infop = png_create_info_struct(pngp);
+  if(!infop) {
+    std::cout << "Failed to make info pointer.";
+    return;
+  }
+
+  png_init_io(pngp, fp);
+
+//  std::cout << "\n\n\nRight before the IDHR!\n\n\n";
+
+  png_set_IHDR (pngp,
+                infop,
+                width,
+                height,
+                depth,
+                (inColor?PNG_COLOR_TYPE_RGB:PNG_COLOR_TYPE_GRAY),
+                PNG_INTERLACE_NONE,
+                PNG_COMPRESSION_TYPE_DEFAULT,
+                PNG_FILTER_TYPE_DEFAULT);
+//  std::cout << "\n\n\nRight before info write!\n\n\n";
+
+  png_write_info(pngp, infop);
+
+//  row = (png_bytep) malloc(width * sizeof(png_byte));
+//std::cout << "\n\n\nRight before setting row!\n\n\n";
+
+  row = new png_byte[(width * 3)+1];
+
+//  Point * p(0);
+//  Point pt(*p);
+//std::cout << "\n\n\nPOINT!\n\n\n";
+
+  Point pt(1,1,0);
+//  std::cout << "\n\n\nvec!\n\n\n";
+  std::vector<double> v ={5};
+//  std::cout << "\n\n\ndv!\n\n\n";
+  DenseVector<Number> dv(0);
+
+//   row = (png_bytep) malloc(width * sizeof(png_byte));
+  // int grayscale[((int)(resolution*resolution))];
+  // for (int i = 0; i < resolution*resolution; i++)
+  //   grayscale[i] = rand() % 256;
```

Comment:
Go ahead and clean up your comments. If you want them around in history feel free to add another commit that removes them that you don't squash. That way they will be in the intermediate commit.